src/uct/ib/base/ib_md.c

```diff
@@ -1411,6 +1451,58 @@ uct_ib_md_parse_subnet_prefix(const char *subnet_prefix_str,
     return UCS_OK;
 }
 
+static double uct_ib_md_read_pci_bw(struct ibv_device *ib_device)
+{
+    const char pci_data_path_fmt[]   = "/sys/class/infiniband/%s/device/%s";
+    const char pci_width_file_name[] = "current_link_width";
+    const char pci_speed_file_name[] = "current_link_speed";
+    char pci_width_str[16];
+    char pci_speed_str[16];
+    uct_ib_md_pci_info_t *p;
+    double bw;
+    unsigned width;
+    ssize_t len;
+    size_t i;
+
+    len = ucs_read_file(pci_width_str, sizeof(pci_width_str) - 1, 1, pci_data_path_fmt,
+                        ib_device->name, pci_width_file_name);
+    if (len < 1) {
+        return DBL_MAX; /* failed to read file */
+    } else {
+        pci_width_str[len] = '\0';
+    }
+
+    len = ucs_read_file(pci_speed_str, sizeof(pci_speed_str) - 1, 1, pci_data_path_fmt,
+                        ib_device->name, pci_speed_file_name);
+    if (len < 1) {
+        return DBL_MAX; /* failed to read file */
+    } else {
+        pci_speed_str[len] = '\0';
+    }
+
+    if (sscanf(pci_width_str, "%u", &width) < 1) {
+        return DBL_MAX;
+    }
+
+    if (sscanf(pci_speed_str, "%lf", &bw) < 1) {
+        return DBL_MAX;
+    }
+
+    bw *= UCS_GBYTE / 8; /* to gigabit */
+
+    for (i = 0; i < ucs_static_array_size(uct_ib_md_pci_info); i++) {
+        if (bw < (uct_ib_md_pci_info[i].bw * 1.2)) { /* use 1.2 multiplex to avoid round issues */
+            p = &uct_ib_md_pci_info[i]; /* use pointer to make equation shorter */
+            return bw * width *
+                   (p->payload * p->nack) /
+                   (((p->payload + p->overhead) * p->nack) + p->ctrl) *
```

Comment:
:+1:

Reply:
let's add ucs_trace like:
`mlx5_0: pcie 8x gen3, effective throughput 13000 MB/s"
(13000 should be replaced by actual number)

Reply:
added trace MB/s and Gb/s (for bench and HCA speed)